modules/machineset-creating-azure-ultra-disk.adoc

```diff
@@ -0,0 +1,292 @@
+
+. In a text editor, open the `userData.txt` file and locate the final `}` character in the file. 
+
+.. On the immediately preceding line, add a `,`. 
+
+.. Create a new line after the `,` and add the following configuration details:
++
+[source,json]
+----
+"storage": {
+  "disks": [ <1>
+    {
+      "device": "/dev/disk/azure/scsi1/lun0", <2>
+      "partitions": [ <3>
+        {
+          "label": "lun0p1", <4>
+          "sizeMiB": 1024, <5>
+          "startMiB": 0
+        }
+      ]
+    }
+  ],
+  "filesystems": [ <6>
+    {
+      "device": "/dev/disk/by-partlabel/lun0p1",
+      "format": "xfs",
+      "path": "/var/lib/lun0p1"
+    }
+  ]
+},
+"systemd": {
+  "units": [ <7>
+    {
+      "contents": "[Unit]\nBefore=local-fs.target\n[Mount]\nWhere=/var/lib/lun0p1\nWhat=/dev/disk/by-partlabel/lun0p1\nOptions=defaults,pquota\n[Install]\nWantedBy=local-fs.target\n",
+      "enabled": true,
+      "name": "var-lib-lun0p1.mount"
+    }
+  ]
+}
+----
+<1> The configuration details for the disk that you want to attach to a node as an ultra disk.
+<2> Specify the `lun` value that is defined in the `dataDisks` stanza of the machine set you are using. For example, if the machine set contains `lun: 0`, specify `lun0`. You can initialize multiple data disks by specifying multiple `"disks"` entries in this configuration file. If you specify multiple `"disks"` entries, ensure that the `lun` value for each matches the value in the machine set.
+<3> The configuration details for a new partition on the disk.
+<4> Specify a label for the partition. You might find it helpful to use hierarchical names, such as `lun0p1` for the first partition of `lun0`.
+<5> Specify the total size in MiB of the partition.
+<6> Specify the filesystem to use when formatting a partition. Use the partition label to specify the partition.
+<7> Specify a `systemd` unit to mount the partition at boot. Use the partition label to specify the partition. You can create multiple partitions by specifying multiple `"partitions"` entries in this configuration file. If you specify multiple `"partitions"` entries, you must specify a `systemd` unit for each.
+
+. Extract the disabling template value to a file called `disableTemplating.txt` by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get secret worker-user-data \
+--template='{{index .data.disableTemplating | base64decode}}' | jq > disableTemplating.txt
+----
+
+. Combine the `userData.txt` file and `disableTemplating.txt` file to create a data secret file by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api create secret generic worker-user-data-x5 \
+--from-file=userData=userData.txt \
+--from-file=disableTemplating=disableTemplating.txt
+----
+endif::mapi[]
+
+. Edit or copy an existing Azure `MachineSet` custom resource (CR) by running the following command:
++
+[source,terminal]
+----
+$ oc edit machineset <machine-set-name>
```

Comment:
I'd drop the edit option and just mention the user should copy/clone an existing MachineSet and update its config to reference the newly created Secret `worker-user-data-x5`.

Reply:
ok cool - backed the indent up one level so that `userDataSecret` and `dataDisks` are peers under `value`